### PR TITLE
bp: Properly document keepalive and other tcp options

### PR DESCRIPTION
**Pull request by BaurzhanSakhariev**

### CODE REVIEW

**Comment by BaurzhanSakhariev** [comment]
There is also a relevant [patch](https://github.com/elastic/elasticsearch/commit/7aeb2fe73cf1f17aee0d4c3c2b81231f6dd7546c) adding some more TCP settings which we might port as well.

File: docs/config/node.rst [hunk 1]
```diff
@@ -98,10 +98,13 @@ Read-only node
   If set to ``true``, the node will only allow SQL statements which are
   resulting in read operations.
 
+Networking
```
Comment (BaurzhanSakhariev) [inline docs/config/node.rst:101]: Re-structuring inspired by https://www.elastic.co/guide/en/elasticsearch/reference/7.10/modules-network.html

Added a new section gather all relevant settings (undocumented before TCP and transport settings goes there as well)

File: docs/config/node.rst [hunk 2]
```diff
@@ -205,6 +208,65 @@ Ports
...
+
+  The size of the TCP receive buffer. By default not explicitly set.
+
+.. NOTE::
+
+    Each setting in this section has its counterpart for HTTP and transport.
+    To provide a protocol specific setting, remove ``network`` prefix and use
+    either ``http`` or ``transport`` instead. For example, no_delay can be
+    configured as ``http.tcp.no_delay`` and ``transport.tcp.no_delay``. Please
+    note, that PG interface takes its settings from transport.
```
Comment (BaurzhanSakhariev) [inline docs/config/node.rst:265]: For HTTP we have `Netty4HttpServerTransport`. Last sentence that PG settings = transport settings is written according to `PgClient` configuration code.

File: docs/config/node.rst [hunk 3]
```diff
@@ -205,6 +208,65 @@ Ports
   range is used. If this is set to an integer value it is considered as an
   explicit single port.
 
+Advanced TCP settings
+---------------------
+
+Any interface that uses TCP (PG/HTTP/transport) shares the following settings:
```
Comment (BaurzhanSakhariev) [inline docs/config/node.rst:214]: relevant `TranportSettings` and `HttpTransportSettings` take their defaults from the `NetworkService` settings.

File: docs/config/node.rst [hunk 4]
```diff
@@ -205,6 +208,93 @@ Ports
...
+  Set to `true` to enable compression (DEFLATE) between all nodes.
+
+.. _transport.ping_schedule:
+
+**transport.ping_schedule**
+  | *Default:* ``-1``
+  | *Runtime:* ``no``
+
+  Schedule a regular application-level ping message to ensure that transport
+  connections between nodes are kept alive. Defaults to `-1` (disabled). It is
```
Comment (BaurzhanSakhariev) [inline docs/config/node.rst:293]: in the referenced version default is either 5s or -1 (transport only) but our version has only -1 default

File: docs/config/node.rst [hunk 5]
```diff
@@ -205,6 +208,93 @@ Ports
   range is used. If this is set to an integer value it is considered as an
   explicit single port.
 
+Advanced TCP settings
+---------------------
+
+Any interface that uses TCP (PG/HTTP/transport) shares the following settings:
```
Comment (matriv) [inline docs/config/node.rst:214]: minor: Personally, I would expand to `Postgres wire, HTTP & Transport protocols`

File: docs/config/node.rst [hunk 6]
```diff
@@ -205,6 +208,93 @@ Ports
...
+   Configures the `` SO_REUSEADDRS`` option for sockets, which determines
+   whether they should reuse the address.
+
+.. _network.tcp.send_buffer_size:
+
+**network.tcp.send_buffer_size**
+  | *Default:* ``-1``
+  | *Runtime:* ``no``
+
+  The size of the TCP send buffer. By default not explicitly set.
```
Comment (matriv) [inline docs/config/node.rst:249]: Can we explain what does that mean? no buffering at all?

File: docs/config/node.rst [hunk 7]
```diff
@@ -205,6 +208,93 @@ Ports
...
+
+.. _network.tcp.receive_buffer_size:
+
+**network.tcp.receive_buffer_size**
+  | *Default:* ``-1``
+  | *Runtime:* ``no``
+
+  The size of the TCP receive buffer. By default not explicitly set.
```
Comment (matriv) [inline docs/config/node.rst:257]: same here.

**Review by matriv — COMMENTED** [review]
Left a few comments, Thank you!

File: docs/config/node.rst (hunk 5 above)
Comment (BaurzhanSakhariev) [reply docs/config/node.rst]: fine with that wording too - pushed a fixup

File: docs/config/node.rst (hunk 7 above)
Comment (BaurzhanSakhariev) [reply docs/config/node.rst]: it's a hint for underlying socket. quote from docs

> In the case of stream-oriented sockets and the TCP/IP protocol, the size of the socket receive buffer may be used when advertising the size of the TCP receive window to the remote peer.

I added a link to socket docs

File: docs/config/node.rst (hunk 6 above)
Comment (BaurzhanSakhariev) [reply docs/config/node.rst]: See comment below. Added link to the corresponding socket option docs

**Review by matriv — APPROVED** [review]
Thank you!!

**Comment by seut** [comment]
> There is also a relevant [patch](https://github.com/elastic/elasticsearch/commit/7aeb2fe73cf1f17aee0d4c3c2b81231f6dd7546c) adding some more TCP settings which we might port as well.

This patch looks interesting and I wonder why it slipped through. Was it marked as skipped?
As I see no super urgency in merging this backport, I'd rather prefer to agree first if a depended older patch should be backported. Otherwise we have to tough the code again and the changes are out-of-order(which may not be a big issue though).

**Comment by BaurzhanSakhariev** [comment]
> Was it marked as skipped?

I see why it slipped through - it's in the list of deferred patches.

I've merged docs PR both into master and 5.1 - I think it should not conflict with the deferred patch as it's only about docs and settings I have documented are kept intact.

**Pull request merged**